var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,176 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Qmcpack(CMakePackage):
+    """QMCPACK, is a modern high-performance open-source Quantum Monte
+       Carlo (QMC) simulation code."""
+
+    # Package information
+    homepage = "http://www.qmcpack.org/"
+    url      = "https://github.com/QMCPACK/qmcpack/archive/v3.1.0.tar.gz"
+
+    version('3.1.0', 'bdf3acd090557acdb6cab5ddbf7c7960')
+    version('3.0.0', '75f9cf70e6cc6d8b7ff11a86340da43d')
+
+    # This download method is untrusted, and is not recommended
+    # by the Spack manual.
+    version('develop', git='https://github.com/QMCPACK/qmcpack.git')
+
+    # These defaults match those in the QMCPACK manual
+    variant('debug', default=False, description='Build debug version')
+    variant('mpi', default=True, description='Build with MPI support')
+    variant('cuda', default=False,
+            description='Enable CUDA and GPU acceleration.')
+    variant('complex', default=True,
+            description='Build the complex (general twist/k-point) version')
+    variant('mixed', default=False,
+            description='Build the mixed precision (mixture of single and '
+                        'double precision) version for gpu and cpu '
+                        '(experimental)')
+    variant('gui', default=False,
+            description='Install with Matplotlib (long installation time)')
+
+    depends_on('cmake@3.4.3:', type='build')
+    depends_on('mpi', when='+mpi')
+    depends_on('libxml2')
+    depends_on('hdf5+mpi', when='+mpi')
+    depends_on('hdf5~mpi', when='~mpi')
```

Comment:
You should also add `depends_on('hdf5')` here. Otherwise, you can't do `spack install qmcpack ^hdf5@1.8.12`. Concretizer bug.